### PR TITLE
check client if nil before revoke

### LINKED ISSUE #77174: kubelet: Observed a panic: "invalid memory address or nil pointer dereference"
**What happened**:

Kubelet startup error

```
E0428 09:37:13.276141   26151 runtime.go:69] Observed a panic: "invalid memory address or nil pointer dereference" (runtime error: invalid memory address or nil pointer dereference)
/workspace/anago-v1.14.1-beta.0.44+b7394102d6ef77/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:76
/workspace/anago-v1.14.1-beta.0.44+b7394102d6ef77/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:65
/workspace/anago-v1.14.1-beta.0.44+b7394102d6ef77/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:51
/usr/local/go/src/runtime/panic.go:522
/workspace/anago-v1.14.1-beta.0.44+b7394102d6ef77/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/client-go/tools/cache/reflector.go:189
/workspace/anago-v1.14.1-beta.0.44+b7394102d6ef77/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/client-go/tools/cache/reflector.go:214
/workspace/anago-v1.14.1-beta.0.44+b7394102d6ef77/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/client-go/tools/cache/reflector.go:125
/workspace/anago-v1.14.1-beta.0.44+b7394102d6ef77/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:152
/workspace/anago-v1.14.1-beta.0.44+b7394102d6ef77/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:153
/workspace/anago-v1.14.1-beta.0.44+b7394102d6ef77/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:88
/workspace/anago-v1.14.1-beta.0.44+b7394102d6ef77/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/client-go/tools/cache/reflector.go:124
/workspace/anago-v1.14.1-beta.0.44+b7394102d6ef77/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:54
/workspace/anago-v1.14.1-beta.0.44+b7394102d6ef77/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:71
/usr/local/go/src/runtime/asm_amd64.s:1337
```

**What you expected to happen**:

**How to reproduce it (as minimally and precisely as possible)**:

```
# yum install -y kubelet
# kubelet --version
Kubernetes v1.14.1
# kubelet --cgroup-driver=systemd --runtime-cgroups=/systemd/system.slice --kubelet-cgroups=/systemd/system.slice --fail-swap-on=false
```

**Anything else we need to know?**:

`kubernetes 1.13.3` does not have this problem.

**Environment**:
- Kubernetes version (use `kubectl version`): 

`Kubernetes v1.14.1`

- OS (e.g: `cat /etc/os-release`):

```
NAME="CentOS Linux"
VERSION="7 (Core)"
```

### PR DESCRIPTION
**Pull request by zouyee**
**What type of PR is this?**
 /kind bug
/sig storage
@kubernetes/sig-storage-bugs

**What this PR does / why we need it**:

Check client if nil before revoke

**Which issue(s) this PR fixes**:

Fixes #77174

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

**Comment by zouyee** [comment]
@kubernetes/sig-storage-bugs

**Review by mattjmcnaughton — COMMENTED** [review]
My initial thought is that this seems like a sensible fix, but curious to get the opinion of @tallclair,
as this seems in a similar vein to https://github.com/kubernetes/kubernetes/pull/77888.

Specifically, @tallclair previous fix focused on us not calling a method that couldn't be successful without a kubeclient, whereas this fix lets us call the method, and just makes it a no-op if there's no kubeclient. I'm not sure if the former is an option in this case or if the project has an opinion on one being better than the other.

**Comment by tedyu** [comment]
I would prefer @tallclair 's approach.
If there is no kubeclient, reconciler shouldn't run.

**Comment by zouyee** [comment]
This `runtime` object depends directly on `kubeClient`, but `volumemanager` is not, they have essential differences, @jsafrane @tallclair WDYT?

File: pkg/kubelet/volumemanager/reconciler/reconciler.go [hunk 1]
```diff
@@ -570,6 +570,10 @@ func (rc *reconciler) reconstructVolume(volume podVolume) (*reconstructedVolume,
 
 // updateDevicePath gets the node status to retrieve volume device path information.
 func (rc *reconciler) updateDevicePath(volumesNeedUpdate map[v1.UniqueVolumeName]*reconstructedVolume) {
+	if rc.kubeClient == nil {
```
Comment (msau42) [inline pkg/kubelet/volumemanager/reconciler/reconciler.go]: I think there is some bigger issue here.  IIUC, this was originally added to address https://github.com/kubernetes/kubernetes/issues/54108.  However, reconciler now only runs once during kubelet initialization, so I'm not sure #54108 is fixed anymore.  Also, in the meantime, we added support for a DeviceMounter interface for plugins that do local attach, which may be able to help address the original issue.

/assign @jingxu97 @mkimuram 

If reconstruction is no longer the right place to fix the issue, then we can remove the call entirely from the reconstruction path.

Comment (mkimuram) [reply pkg/kubelet/volumemanager/reconciler/reconciler.go]: As I checked the current code, the original issue discussed in #54108, or `devicePath` isn't set properly for non-attachable plugin, is resolved by passing `devicePath` to `MarkDeviceAsMounted` after `WaitForAttach` in `GenerateMountVolumeFunc` and `GenerateMapVolumeFunc`.
(There seemed a discussion to solve this issue by updating it there, though.)

I think that this code is only for volume reconstruction and was originally introduced in https://github.com/kubernetes/kubernetes/pull/33616. And, it seems to be still needed for this purpose.

@jingxu97

Is my understanding right?

**Pull request closed without merging**